src/cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -423,17 +428,33 @@ function redirect3ds2() {
   const environment = AdyenHelper.getAdyenEnvironment().toLowerCase();
   const locale = request.getLocale();
 
+  const orderNo = request.httpParameterMap.get('merchantReference').stringValue;
+  const orderToken = request.httpParameterMap.get('orderToken').stringValue;
+  const order = OrderMgr.getOrder(orderNo, orderToken);
+  const paymentInstrument = order.getPaymentInstruments(
+      constants.METHOD_ADYEN_COMPONENT,
+  )[0];
+  const action = paymentInstrument.custom.adyenAction;
+  Transaction.wrap(function () {
+    paymentInstrument.custom.adyenAction = null;
+  });
+
   app
     .getView({
       locale,
       originKey,
       environment,
       resultCode: request.httpParameterMap.get('resultCode').stringValue,
-      token3ds2: request.httpParameterMap.get('token3ds2').stringValue,
-      merchantReference: request.httpParameterMap.get('merchantReference').stringValue,
+      action: action,
+      merchantReference: orderNo,
+      orderToken: orderToken,
       ContinueURL: URLUtils.https(
         'Adyen-Authorize3DS2',
```

Comment:
is there a plan to move to a generic url that handles 3DS2 and the other payments/details requests?

Reply:
Recently we started using create from action for 3DS2 payments, the next step would be moving to a generic URL